src/helpers/makeQueries.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+import waitFor from '../waitFor';
+import type { WaitForOptions } from '../waitFor';
+import { ErrorWithStack } from './errors';
+
+type QueryFunction<ArgType, ReturnType> = (
+  instance: ReactTestInstance
+) => (args: ArgType) => ReturnType;
+
+type FindQueryFunction<ArgType, ReturnType> = (
+  instance: ReactTestInstance
+) => (args: ArgType, waitForOptions?: WaitForOptions) => Promise<ReturnType>;
+
+type QueryAllByQuery<QueryArg> = QueryFunction<
+  QueryArg,
+  Array<ReactTestInstance>
+>;
+type QueryByQuery<QueryArg> = QueryFunction<QueryArg, null | ReactTestInstance>;
+
+type GetAllByQuery<QueryArg> = QueryFunction<
+  QueryArg,
+  Array<ReactTestInstance>
+>;
+type GetByQuery<QueryArg> = QueryFunction<QueryArg, ReactTestInstance>;
+
+type FindAllByQuery<QueryArg> = FindQueryFunction<
+  QueryArg,
+  Array<ReactTestInstance>
+>;
+type FindByQuery<QueryArg> = FindQueryFunction<QueryArg, ReactTestInstance>;
+
+export type Queries<QueryArg> = {
+  getBy: GetByQuery<QueryArg>,
+  getAllBy: GetAllByQuery<QueryArg>,
+  queryBy: QueryByQuery<QueryArg>,
+  findBy: FindByQuery<QueryArg>,
+  findAllBy: FindAllByQuery<QueryArg>,
+};
```

Comment:
@MattAgn I made Queries parametrized with QueryArg, should solve the problem of queries having different args.

Reply:
Nice! I tried it as well but couldn't make it work with the spread operator if we were to add options for example, or simply to make it work for the find queries, which also have waitForOptions